From efd533f644e8f224d75c86adc3d8937e05afb42e Mon Sep 17 00:00:00 2001
From: Caleb Connolly <caleb@connolly.tech>
Date: Sat, 8 May 2021 09:35:07 +0100
Subject: [PATCH 40/44] arm64: dts: qcom: sdm845-oneplus: fix reserved-mem

Fix the upper guard and the "removed_region", this fixes the random
crashes which used to occur in memory intensive loads. I'm not sure WHY
the upper guard being 0x2000 instead of 0x1000 doesn't fix this, but it
HAS to be 0x1000.

Signed-off-by: Caleb Connolly <caleb@connolly.tech>
Fixes: e60fd5ac1f68 ("arm64: dts: qcom: sdm845-oneplus-common: guard
rmtfs-mem")
---
 arch/arm64/boot/dts/qcom/sdm845-oneplus-common.dtsi | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/sdm845-oneplus-common.dtsi b/arch/arm64/boot/dts/qcom/sdm845-oneplus-common.dtsi
index e24c80f73e1d..861e71783f76 100644
--- a/arch/arm64/boot/dts/qcom/sdm845-oneplus-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/sdm845-oneplus-common.dtsi
@@ -71,7 +71,7 @@ rmtfs_mem: memory@f5b01000 {
 		};
 		rmtfs_upper_guard: memory@f5d01000 {
 			no-map;
-			reg = <0 0xf5d01000 0 0x2000>;
+			reg = <0 0xf5d01000 0 0x1000>;
 		};
 
 		/*
@@ -80,7 +80,7 @@ rmtfs_upper_guard: memory@f5d01000 {
 		 */
 		removed_region: memory@88f00000 {
 			no-map;
-			reg = <0 0x88f00000 0 0x200000>;
+			reg = <0 0x88f00000 0 0x1c00000>;
 		};
 
 		ramoops: ramoops@ac300000 {
-- 
2.30.2

